src/plugins/intel_cpu/src/nodes/pooling.cpp

```diff
@@ -174,45 +174,49 @@ Pooling::Pooling(const std::shared_ptr<ov::Node>& op, const GraphContext::CPtr c
+    }
 
+    if (auto maxPoolOp_v14 = ov::as_type_ptr<const ov::op::v14::MaxPool>(op)) {
+        get_attributes(poolingAttrs.dilation, maxPoolOp_v14->get_dilations());
+    } else if (auto maxPoolOp_v8 = ov::as_type_ptr<const ov::op::v8::MaxPool>(op)) {
+        isMaxPool8 = true;
         get_attributes(poolingAttrs.dilation, maxPoolOp_v8->get_dilations());
-        get_attributes(poolingAttrs.stride, maxPoolOp_v8->get_strides());
-        get_attributes(poolingAttrs.kernel, maxPoolOp_v8->get_kernel());
-        get_attributes(poolingAttrs.data_pad_begin, maxPoolOp_v8->get_pads_begin());
-        get_attributes(poolingAttrs.data_pad_end, maxPoolOp_v8->get_pads_end());
-
-        poolingAttrs.auto_pad = (maxPoolOp_v8->get_auto_pad() == ov::op::PadType::SAME_LOWER || maxPoolOp_v8->get_auto_pad() == ov::op::PadType::SAME_UPPER);
     } else if (auto maxPoolOp_v1 = ov::as_type_ptr<const ov::op::v1::MaxPool>(op)) {
-        algorithm = Algorithm::PoolingMax;
-        poolingAttrs.exclude_pad = false;
-        poolingAttrs.pad_type = maxPoolOp_v1->get_auto_pad();
-        poolingAttrs.rounding = maxPoolOp_v1->get_rounding_type();
+        poolingAttrs.dilation.resize(poolingAttrs.kernel.size(), 1);
+    } else if (auto avgPoolOp_v1 = ov::as_type_ptr<const ov::op::v1::AvgPool>(op)) {
+        algorithm = Algorithm::PoolingAvg;
+        poolingAttrs.exclude_pad = avgPoolOp_v1->get_exclude_pad();
+        poolingAttrs.rounding = avgPoolOp_v1->get_rounding_type();
 
-        get_attributes(poolingAttrs.stride, maxPoolOp_v1->get_strides());
-        get_attributes(poolingAttrs.kernel, maxPoolOp_v1->get_kernel());
-        get_attributes(poolingAttrs.data_pad_begin, maxPoolOp_v1->get_pads_begin());
-        get_attributes(poolingAttrs.data_pad_end, maxPoolOp_v1->get_pads_end());
+        get_attributes(poolingAttrs.stride, avgPoolOp_v1->get_strides());
+        get_attributes(poolingAttrs.kernel, avgPoolOp_v1->get_kernel());
+        get_attributes(poolingAttrs.data_pad_begin, avgPoolOp_v1->get_pads_begin());
+        get_attributes(poolingAttrs.data_pad_end, avgPoolOp_v1->get_pads_end());
         poolingAttrs.dilation.resize(poolingAttrs.kernel.size(), 1);
 
-        poolingAttrs.auto_pad = (maxPoolOp_v1->get_auto_pad() == ov::op::PadType::SAME_LOWER || maxPoolOp_v1->get_auto_pad() == ov::op::PadType::SAME_UPPER);
-    } else if (auto avgPoolOp = ov::as_type_ptr<const ov::op::v1::AvgPool>(op)) {
+        poolingAttrs.auto_pad = (avgPoolOp_v1->get_auto_pad() == ov::op::PadType::SAME_LOWER || avgPoolOp_v1->get_auto_pad() == ov::op::PadType::SAME_UPPER);
+    } else if (auto avgPoolOp_v14 = ov::as_type_ptr<const ov::op::v14::AvgPool>(op)) {
         algorithm = Algorithm::PoolingAvg;
-        poolingAttrs.exclude_pad = avgPoolOp->get_exclude_pad();
-        poolingAttrs.rounding = avgPoolOp->get_rounding_type();
+        poolingAttrs.exclude_pad = avgPoolOp_v14->get_exclude_pad();
+        poolingAttrs.rounding = avgPoolOp_v14->get_rounding_type();
```

Comment:
Where the `ceil_torch` is processed? Please check generated input data in tests to be sure that such case is covered.

Reply:
I haven't yet added tests for the new functionality, first I'm working on the original tests working with new operator versions. I'm constantly encountering errors such as:

```bash
/home/pwysocki/openvino/src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp:331: Failure
Expected: (it) != (inputMap.end()), actual: 8-byte object <78-C0 8C-E1 FF-7F 00-00> vs 8-byte object <78-C0 8C-E1 FF-7F 00-00>
/home/pwysocki/openvino/src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp:95: Failure
Exception from src/inference/src/cpp/infer_request.cpp:223:
Exception from src/plugins/intel_cpu/src/node.cpp:1620:
Shape inference of MaxPool node with name MaxPool_88923 failed: Check 'batch_size.is_dynamic() || batch_size.get_length() > 0' failed at src/core/shape_inference/include/pooling_shape_inference_util.hpp:277:
While validating node 'opset8::MaxPool MaxPool_88923 (opset1::Parameter Parameter_88922[0]:f32[1..5,4..32,1..64,1..64,1..25]) -> (f32[1..5,4..32,1..33,2..33,1..13], i32[1..5,4..32,1..33,2..33,1..13])' with friendly_name 'MaxPool_88923':
Batch size is zero.
``` 

or 

```bash
/home/pwysocki/openvino/src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp:331: Failure
Expected: (it) != (inputMap.end()), actual: 8-byte object <78-C0 8C-E1 FF-7F 00-00> vs 8-byte object <78-C0 8C-E1 FF-7F 00-00>
/home/pwysocki/openvino/src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp:95: Failure
Couldn't find input with name Parameter_82447 in the inputs map
```

Reply:
The tests have been added.

Reply:
I still don't see any processing of the `ceil_torch` in the plugin, so tests with this attribute must fail. Most likely you need to adjust generated input data in test.
Are you going to implement that in this PR? If not, need to disabled such tests [here](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/tests/functional/shared_tests_instances/skip_tests_config.cpp) until it will be implemented.
As I understand it, this is the only difference from the previous version of operation?